### ISSUE TITLE
Where is params?  undefined local variable or method `params'  IssueTracker

### ISSUE THREAD

**Issue by nofxx** [brandedcrate/errbit_gitlab_plugin#1]
When updating an App trying to add Gitlab issue tracker:

```
undefined local variable or method `params' for #<ErrbitGitlabPlugin::IssueTracker:0x00000007b6b708>

errbit_gitlab_plugin (0.1.0) lib/errbit_gitlab_plugin/issue_tracker.rb:67:in `block in errors'
errbit_gitlab_plugin (0.1.0) lib/errbit_gitlab_plugin/issue_tracker.rb:67:in `each'
errbit_gitlab_plugin (0.1.0) lib/errbit_gitlab_plugin/issue_tracker.rb:67:in `detect'
errbit_gitlab_plugin (0.1.0) lib/errbit_gitlab_plugin/issue_tracker.rb:67:in `errors'
app/models/issue_tracker.rb:22:in `validate_tracker'
```

```
#67  if self.class.fields.detect {|f| params[f[0]].blank? }
```

Trying to fix it but failing miserably, any advice?

**Comment by gabrielhilal** [comment]
The same issue here...

**Comment by kohenkatz** [comment]
`params` needs to be changed to `options`, but some other changes are also needed.

See https://github.com/errbit/errbit_github_plugin/commit/5900200e6d460fe94feaea3e59824437d54f1029 for the changes that were needed to the GitHub plugin.

**Comment by gabrielhilal** [comment]
BTW, I have solved the problem using the errbit branch 0-3-stable, which includes all the integrations...

**Comment by kohenkatz** [comment]
Eventually, the plugin will need to be updated for the new version anyway...

**Comment by stevecrozz** [comment]
Yeah. We've updated the interface and we need a corresponding change in this plugin analogous to https://github.com/brandedcrate/errbit_pivotal_plugin/pull/2

Anyone up for it?

**Comment by stex** [comment]
I updated the plugin to work with the latest errbit_plugin version and submitted a pull request (#3)

@stevecrozz Could you clarify the usage of own templates in issue trackers?

**Comment by stevecrozz** [comment]
@Stex I think your question is about custom templates on issue trackers. Here's how that works. When creating an issue from Errbit, we call [render_to_string](https://github.com/errbit/errbit/blob/v0.4.0/app/controllers/problems_controller.rb#L60) using arguments [supplied by the tracker (or defaults)](https://github.com/errbit/errbit/blob/v0.4.0/app/models/issue.rb#L13-L19).

So your tracker can implement `render_body_args` and the return value will be sent to `render_to_string` and the result is the body of the issue. It looks like this is not documented in errbit_plugin. I'll raise an issue about that so I don't forget: https://github.com/errbit/errbit_plugin/issues/5

**Comment by stex** [comment]
I'm closing this now as it should be fixed with the gem update to v1.0.0
